Frends.Community.Email/EmailTask.cs

```diff
@@ -122,6 +130,130 @@ public static Output SendEmail([PropertyTab]Input message, [PropertyTab]Attachme
+        /// </summary>
+        /// <returns>
+        /// Object { bool EmailSent, string StatusString }
+        /// </returns>
+        public static async Task<Output> SendEmailToExchangeServer([PropertyTab] ExchangeInput input, [PropertyTab] Attachment[] attachments, [PropertyTab] ExchangeServer settings, CancellationToken cancellationToken)
+        {
+            var output = new Output();
+
+            if (string.IsNullOrWhiteSpace(settings.AppId) || string.IsNullOrWhiteSpace(settings.TenantId) || string.IsNullOrWhiteSpace(settings.Username) || string.IsNullOrWhiteSpace(settings.Password))
+                throw new ArgumentException("Invalid Application ID, Tenant ID, Username or Password. Please check Exchange settings.");
+
+            if (string.IsNullOrWhiteSpace(input.Subject) || string.IsNullOrWhiteSpace(input.Message) || string.IsNullOrWhiteSpace(input.To))
+                throw new ArgumentException("Subject, message, and To-recipient cannot be empty.");
+
+            var options = new TokenCredentialOptions
+            {
+                AuthorityHost = AzureAuthorityHosts.AzurePublicCloud
+            };
+
+            var credentials = new UsernamePasswordCredential(settings.Username, settings.Password, settings.TenantId, settings.AppId, options);
+            var graph = new GraphServiceClient(credentials);
+
+            var encoding = GetEncoding(input.MessageEncoding);
+            var bytes = Encoding.Default.GetBytes(input.Message);
+            var encodedmessage = encoding.GetString(bytes);
+
+            var messageBody = new ItemBody
+            {
+                ContentType = input.IsMessageHtml ? BodyType.Html : BodyType.Text,
+                Content = encodedmessage
+            };
+
+            var recipients = input.To.Split(new char[] { ',', ';'}, StringSplitOptions.RemoveEmptyEntries);
+
+            var to = new List<Recipient>();
+            var cc = new List<Recipient>();
+            var bcc = new List<Recipient>();
+
+            foreach (var receiver in recipients) to.Add(new Recipient { EmailAddress = new EmailAddress { Address = receiver } });
+
+            if (!string.IsNullOrWhiteSpace(input.Cc))
+            {
+                recipients = input.Cc.Split(new char[] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries);
+                foreach (var receiver in recipients) cc.Add(new Recipient { EmailAddress = new EmailAddress { Address = receiver } });
+            }
+            if (!string.IsNullOrWhiteSpace(input.Bcc))
+            {
+                recipients = input.Bcc.Split(new char[] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries);
+                foreach (var receiver in recipients) bcc.Add(new Recipient { EmailAddress = new EmailAddress { Address = receiver } });
+            }
+
+            var attachmentList = new MessageAttachmentsCollectionPage();
+
+            if (attachments != null)
```

Comment:
This is a huge method. Can we possibly split it up a bit and make more readable by e.g. taking larger chunks into own helpers. E.g. attachments to own method or something?..